App/Repositories/EloquentRepositoryInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Repositories;
+
+
+use Illuminate\Database\Eloquent\Model;
+
+/**
+* Interface EloquentRepositoryInterface
+* @package App\Repositories
+*/
+interface EloquentRepositoryInterface
+{
+
+    public function getAll();
+
+    public function find($id);
+
+    public function findBy($field_name, $field_value);
+
+    public function findAllBy($field_name, $field_value);
+
+    //public function getBasicQuery(): Builder;
```

Comment:
you can use this on all repositories to return this  as an example:

```
    /**
     * @return Builder
     */
    public function getBasicQuery(): Builder
    {
        return User::with($this->with);
    }
```

Reply:
done